RecoTracker/TkTrackingRegions/src/OuterHitPhiPrediction.h

```diff
@@ -4,32 +4,40 @@
 /** predicts phi range at a given radius r */
 
 #include "RecoTracker/TkMSParametrization/interface/PixelRecoRange.h"
-#include "RecoTracker/TkTrackingRegions/interface/TkTrackingRegionsMargin.h"
 
 #include "FWCore/Utilities/interface/Visibility.h"
 
+#include <cassert>
+
 class dso_internal OuterHitPhiPrediction {
 public:
 
-  typedef PixelRecoRange<float> Range;
-  typedef TkTrackingRegionsMargin<float> Margin;
+  using Range= PixelRecoRange<float>;
 
   OuterHitPhiPrediction( 
       const Range & phiAtVertex, 
       const Range & curvature, 
-      float originRBound, 
-      const Margin & tolerance = Margin(0,0))
+      float originRBound) 
     : thePhiAtVertex(phiAtVertex), theCurvature(curvature),
-      theOriginRBound (originRBound), theTolerance(tolerance) { } 
+      theOriginRBound (originRBound) {
+       assert(theCurvature.max()>0);
```

Comment:
@fwyzard @Martin-Grunewald
To further clarify: specify a snapshot in a GT make sense only when the underlying condition set is 'a good one' and is frozen. In practice, the GT with snapshot are targeted to specific (non-evolving) data sets. The usage for data 'in the future' wrt the snapshot is in general inappropriate.
That said, we could add a warning to remark the usage outside the snapshot boundary, although this doesn't necessarily consists in something wrong in itself...
It's more about to set up a policy and how strict we want to be ( in the usage of GT with snapshot ). We might even decide to 'close the iovs' and generate an error.